modules/database/audit-autoloaded-options/hooks.php

```diff
@@ -20,9 +20,119 @@
 		'test'  => 'perflab_aao_autoloaded_options_test',
 	);
 	return $tests;
 }
 add_filter( 'site_status_tests', 'perflab_aao_add_autoloaded_options_test' );
+
+/**
+ * Register custom REST API route for updating autoload.
+ *
+ * @since n.e.x.t
+ */
+function perflab_aao_register_rest_route() {
+	register_rest_route(
+		'perflab-aao/v1',
+		'/update-autoload',
+		array(
+			'methods'  => 'POST',
+			'callback' => 'perflab_aao_update_autoload_rest',
+		)
+	);
+}
+add_action( 'rest_api_init', 'perflab_aao_register_rest_route' );
+
+/**
+ * Callback for updating autoload via REST API.
+ *
+ * @since n.e.x.t
+ *
+ * @param WP_REST_Request $request Request object.
+ * @return WP_REST_Response
+ */
+function perflab_aao_update_autoload_rest( $request ) {
+	$option_name = $request->get_param( 'option_name' );
+	$autoload    = $request->get_param( 'autoload' );
+	$value       = $request->get_param( 'value' );
+
+	if ( empty( $option_name ) || empty( $autoload ) ) {
+		return new WP_REST_Response(
+			array(
+				'success' => false,
+				'message' => esc_html__( 'Invalid option name or autoload value.', 'performance-lab' ),
+			),
+			400
+		);
+	}
+
+	// Check if the option exists.
+	if ( false === get_option( $option_name ) ) {
+		// Option doesn't exist, return an error or handle the situation accordingly.
+		return new WP_REST_Response(
+			array(
+				'success' => false,
+				'message' => esc_html__( 'The option does not exist.', 'performance-lab' ),
+			),
+			400
+		);
+	}
+
+	$result = wp_set_option_autoload( $option_name, $autoload );
+
+	if ( $result ) {
+		// Update modified options list.
+		$modified_options = get_option( 'perflab_aao_modified_options', array() );
+		if ( 'yes' === $autoload && isset( $modified_options[ $option_name ] ) ) {
+			unset( $modified_options[ $option_name ] );
+		} else {
+			$modified_options[ $option_name ] = $value;
+		}
+		update_option( 'perflab_aao_modified_options', $modified_options );
+
+		return new WP_REST_Response(
+			array(
+				'success' => true,
+				'message' => sprintf(
+					/* translators: 1: Autoload value, 2: Option name */
+					esc_html__( 'Autoload %1$s for option %2$s.', 'performance-lab' ),
+					$autoload,
+					$option_name
+				),
+			),
+			200
+		);
+	} else {
+		return new WP_REST_Response(
+			array(
+				'success' => false,
+				'message' => esc_html__( 'Failed to update autoload status.', 'performance-lab' ),
+			),
+			500
+		);
+	}
+}
+
+/**
+ * Enqueue JavaScript for disabling autoload.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $hook_suffix The current admin page.
+ */
+function perflab_aao_enqueue_script( $hook_suffix ) {
+	if ( 'site-health.php' !== $hook_suffix ) {
+		return;
+	}
+	wp_enqueue_script( 'perflab-aao-update-autoload', plugin_dir_url( __FILE__ ) . 'update-autoload.js', array( 'jquery' ), 'n.e.x.t', true );
+
+	wp_localize_script(
+		'perflab-aao-update-autoload',
+		'perflabAutoloadSettings',
+		array(
+			'root' => sanitize_url( get_rest_url() ),
```

Comment:
I don't think this is needed because it's coming straight from WP:

```suggestion
			'root' => get_rest_url(),
```